sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -618,8 +619,11 @@ public void popScope() {
           .getLogger()
           .log(SentryLevel.WARNING, "Instance is disabled and this 'popScope' call is a no-op.");
     } else {
-      // TODO how to remove fork?
-      // TODO getParentScopes().makeCurrent()?
+      final @Nullable Scopes parent = getParent();
+      if (parent != null) {
+        // TODO this is never closed
+        parent.makeCurrent();
```

Comment:
We're handed back a lifecycle token here which we never call close on. Popping scopes this way is kinda hacky. I haven't come up with a better aproach yet.

Reply:
The previous `Scopes` should be restored, when the current one is closed, so we can't close it when creating a new (child) scope.

Take a piece of nested code:

```
// outermost scopes (scopes1)
Sentry.withScope(scope -> {
	// inner scopes (scopes2)
	try { ... = scope.forkedCurrentScopes("...").makeCurrent()) {
		// innermost scopes (scopes3)
	}
}
```

When leaving the `try` block, we're closing `scopes3` and restoring `scopes2`. When leaving the `withScope` lambda, we're closing `scopes2` and restoring `scopes1`.

Reply:
I copied this pattern from OpenTelemetry, as we'll have to call their `io.opentelemetry.context.Scope.close()` inside our OTEL flavor lifecycle token. See https://github.com/getsentry/sentry-java/pull/3285/files#diff-ce5e1c852e10adfae80110c82615eaa9571a7eeb5ed8bb79512c08dc388c4bc0R43

Reply:
Think of this chain of lifecycle tokens as a replacement for the stack. We're keeping scopes from being garbage collected by holding a reference to each parent scopes instance.